ax/core/search_space.py

```diff
@@ -744,37 +721,95 @@ def __init__(
         super().__init__(
             parameters=parameters, parameter_constraints=parameter_constraints
         )
-        # NOTE: We do not support env var and input noise together since the existing
-        # acqfs are designed to work with only one of these.
+        self._validate_distributions()
+
+    def _validate_distributions(self) -> None:
+        r"""Validate the parameter distributions.
+
+        * All distributional parameters must be range parameters.
+        * All environmental variables must have a non-multiplicative distribution.
+        * Either all or none of the perturbation distributions must be
+        multiplicative.
+        * Each parameter can have at most one distribution associated with it.
+        """
+        distributions = self.parameter_distributions
+        # Make sure that there is at most one distribution per parameter.
+        dist_params_list = []
+        for dist in distributions:
+            dist_params_list.extend(dist.parameters)
+        self._distributional_parameters: Set[str] = set(dist_params_list)
+        if len(self._distributional_parameters) != len(dist_params_list):
+            raise UserInputError(
+                "Received multiple parameter distributions for at least one parameter. "
+                "Make sure that there is at most one distribution specified for any "
+                "given parameter / environmental variable."
+            )
```

Comment:
Can raise a more informative error here
```suggestion
        self._distributional_parameters: Set[str] = set()
        for dist in distributions:
            duplicates = self._distributional_parameters.intersection(dist.parameters)
            if duplicates:
                raise UserInputError(
                    f"Received multiple parameter distributions for parameters {duplicates}. "
                    "Make sure that there is at most one distribution specified for any "
                    "given parameter / environmental variable."
                )
            self._distributional_parameters.update(dist.parameters)
```